localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -283,12 +283,28 @@ def get_records(self, shard_iterator: str) -> dict:
                 )
                 raise CustomerInvocationError from e
             elif "ResourceNotFoundException" in str(e):
-                LOG.warning(
-                    "Source stream %s does not exist: %s",
+                # FIXME: The 'Invalid ShardId in ShardIterator' error is returned by DynamoDB-local. Unsure when/why this is returned.
+                #
+                if "Invalid ShardId in ShardIterator" in str(e):
+                    LOG.warning(
+                        "Invalid ShardId in ShardIterator for %s. Re-initializing shards.",
+                        self.source_arn,
+                    )
+                    self.initialize_shards()
+                else:
+                    LOG.warning(
+                        "Source stream %s does not exist: %s",
+                        self.source_arn,
+                        e,
+                    )
+                    raise CustomerInvocationError from e
+            elif "TrimmedDataAccessException" in str(e):
```

Comment:
thank you so much for adding this! 🙏 I'll tell the users next week about it for them to confirm to us it solves the issue and they don't need to run the command manually anymore! Thank you very much for going the extra mile here 🙇